### ISSUE TITLE
Validate against JSON schema

### ISSUE THREAD

**Issue by KamasamaK** [foundeo/cfdocs#569]
We should run all of the JSON data files through the [JSON schema](https://github.com/foundeo/cfdocs/blob/master/data/cfdocs.schema.json) to validate them. Once complete, the JSON schema should be integrated into the Travis CI build process so that all future contributions are guaranteed to be valid.

**Comment by pfreitag** [comment]
FYI - I have added this to the travis-ci build, here is an example: https://travis-ci.org/foundeo/cfdocs/jobs/285772841#L320

It does not fail the tests if the schema is invalid right now, it looks like there are a bunch of minor issues to fix before we start doing that.

**Comment by KamasamaK** [comment]
I had gone through several before submitting the schema, and all of the issues I found were that the `type` was not matching the `enum`s provided. This was mostly due to casing, which should be simple to fix. I can go through them a chunk at a time and fix the issues while trying to avoid conflicts.

**Comment by pfreitag** [comment]
That would be great thanks ... it is probably something you could write a script to do, just check the diff's to make sure your pattern doesn't replace part of an example or something.

fileRead(jsonFile).replace(':"String"', ':"string"', "ALL")

**Comment by KamasamaK** [comment]
I came across an issue with validating tags. I may have to modify the schema based on the answer to this question. For tags, the `returns` key:
1. Should not exist
2. Should be the empty string
3. Should be `void`

Option 2 is the only one that would require a schema change, and it is also how all of the files I've look at seem to be set. I would vote option 1 myself.

**Comment by shaedrich** [comment]
Me to :+1:

**Comment by pfreitag** [comment]
Option 1 probably makes the most sense.

**Comment by KamasamaK** [comment]
I will remove the `returns` from tag docs based on the feedback.

Another issue I've come across is whether the `default` and `values` items in `params` should match the param's `type`. In [the README example](https://github.com/foundeo/cfdocs/blob/master/README.md#json-file-documentation), I noticed that the `type` was "boolean" and the `default` was a string, so I made that schema under the assumption that it would always just be a string and not necessarily match the type. I've noticed docs that use booleans or numbers instead of strings for those fields when the param's type is "boolean" or "numeric", respectively. There may be other cases I haven't come across, but those are the only ones I've seen. 

I'm not aware of a way to have the schema use conditional types, so the options are that the schema type for those fields:

1. Stays as `string`
2. Only contains simple types: `["string", "number", "boolean"]`
3. Has no restriction

**Comment by pfreitag** [comment]
Good question - I am ok with option 1 or 2. I would lean towards 1 I think but could be convinced otherwise.
